### PR TITLE
Remove category

### PR DESCRIPTION
**Pull request by promact-karan**
Fixed Issues
#20927, #20914, #21183

Files Changed

CategoryController.cs : (CategoryRemove method Added) to delete a category.

ICategoryRepository.cs :  RemoveCategoryToDatabase method declared.

CategoryRepository.cs : RemoveCategoryToDatabase method implemented to remove a category from Category Model.

questions-dashboard.component.ts:  In deleteCategoryDialog method I set the property of DeleteCategoryDialogComponent class. 

delete-category-dialog.html : call removeCategoryData method on yes button.

questions-dashboard.html : call deleteCategoryDialog  method on delete popup menu.

categories.service.ts: (removeCategory method added) to send http request.

delete-category-dialog.component.ts : (removeCategoryData method added) to call removeCategory method of CategoryService class

CategoryRepositoryTest :  DeleteCategory method added for unit testing.

Checks

- [X] Naming Conventions

- [X] Server side code comments (proper English, grammar and no spelling mistakes)

- [X] Client side code comments (proper English, grammar and no spelling mistakes)

- [X] Unused variables, methods, blank spaces and name spaces.

- [X] Optimal Code and code formatting

- [X] Commit History is proper, no merge is done.

- [ ] All test cases are executed provided by tester.

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/categories.service.ts [hunk 1]
```diff
@@ -14,4 +14,8 @@ export class CategoryService {
     getAllCategories() {
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/categories.service.ts]: please make file name singular

Comment (promact-karan) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/categories.service.ts]: I change the method name

Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/categories.service.ts]: i asked you to change file name not method name.

Comment (promact-karan) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/categories.service.ts]: which file

Comment (promact-karan) [inline Trappist/src/Promact.Trappist.Web/wwwroot/app/questions/categories.service.ts]: I changed file name.

**Comment by Krunal-Promact** [comment]
Pull request is not proper.

**Pull request closed without merging**